# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1684958498 -7200
#      Wed May 24 22:01:38 2023 +0200
# Node ID 3598c46d6b58a9e502bed5b395848c2aa8725edd
# Parent  6b457f432bc9ea8bddf6222e9414b8378155e1c8
Backed out changeset ed0ae0da182e: Bug 1829911 - beta appeal. 

diff --git a/mail/app/profile/all-thunderbird.js b/mail/app/profile/all-thunderbird.js
--- a/mail/app/profile/all-thunderbird.js
+++ b/mail/app/profile/all-thunderbird.js
@@ -1337,14 +1337,8 @@ pref("print.print_headerright", "");
 // Enable Masonry Layout for AddressBook.
 pref("layout.css.grid-template-masonry-value.enabled", true);
 
 // End of year donation appeal.
 pref("app.donation.eoy.version", 1);
 pref("app.donation.eoy.version.viewed", 0);
 pref("app.donation.eoy.url", "https://www.thunderbird.net/thunderbird/102.0/eoy/");
 
-// Beta appeal
-pref("app.beta_appeal.version", 1);
-pref("app.beta_appeal.version.viewed", 0);
-pref("app.beta_appeal.url", "https://www.thunderbird.net/thunderbird/102.0/beta-appeal/?locale=%LOCALE%&version=%VERSION%&channel=%CHANNEL%&os=%OS%&buildid=%APPBUILDID%");
-// June 21st expiry.
-pref("app.beta_appeal.expiry", 1687323600);
diff --git a/mail/base/content/msgMail3PaneWindow.js b/mail/base/content/msgMail3PaneWindow.js
--- a/mail/base/content/msgMail3PaneWindow.js
+++ b/mail/base/content/msgMail3PaneWindow.js
@@ -735,21 +735,16 @@ var gMailInit = {
     Services.obs.notifyObservers(window, "mail-startup-done");
 
     // Show the end of year donation appeal page.
     if (this.shouldShowEOYDonationAppeal()) {
       // Add a timeout to prevent opening the browser immediately at startup.
       setTimeout(this.showEOYDonationAppeal, 2000);
     }
 
-    if (AppConstants.platform == "win" && this.shouldShowBetaAppeal()) {
-      // A little later than donation appeal.
-      setTimeout(this.showBetaAppeal, 5000);
-    }
-
     // Idle dispatch the telemetry reports.
     Services.tm.idleDispatchToMainThread(() => {
       reportAccountTypes();
       reportAddressBookTypes();
       reportAccountSizes();
       reportPreferences();
       reportUIConfiguration();
     });
@@ -822,84 +817,16 @@ var gMailInit = {
     let messenger =
       window.messenger ||
       Cc["@mozilla.org/messenger;1"].createInstance(Ci.nsIMessenger);
     messenger.launchExternalURL(url);
 
     let currentEOY = Services.prefs.getIntPref("app.donation.eoy.version", 1);
     Services.prefs.setIntPref("app.donation.eoy.version.viewed", currentEOY);
   },
-
-  /**
-   * Open the beta appeal in a new window to ensure the user can download it
-   * correctly.
-   */
-  showBetaAppeal() {
-    let url = Services.urlFormatter.formatURLPref("app.beta_appeal.url");
-    let messenger =
-      window.messenger ||
-      Cc["@mozilla.org/messenger;1"].createInstance(Ci.nsIMessenger);
-    messenger.launchExternalURL(url);
-
-    this.markBetaAppealVersion();
-  },
-
-  /**
-   * Check if we can trigger the opening of the beta appeal page.
-   *
-   * @returns {boolean} - True if the beta appeal page should be opened.
-   */
-  shouldShowBetaAppeal() {
-    let currentBetaAppeal = Services.prefs.getIntPref(
-      "app.beta_appeal.version",
-      1
-    );
-    let viewedBetaAppeal = Services.prefs.getIntPref(
-      "app.beta_appeal.version.viewed",
-      0
-    );
-
-    // True if the user never saw the beta appeal, this is not a new
-    // profile, and we're not running tests.
-    const applicable =
-      viewedBetaAppeal < currentBetaAppeal &&
-      !specialTabs.shouldShowPolicyNotification() &&
-      !Cu.isInAutomation;
-    if (!applicable) {
-      return false;
-    }
-
-    // expiry date
-    const now = Math.floor(Date.now() / 1000);
-    const expiry = Services.prefs.getIntPref("app.beta_appeal.expiry", now);
-
-    // Never shown, filter to 10% of users.
-    const factor = Math.random() * 100;
-    if (factor > 10 || now > expiry) {
-      this.markBetaAppealVersion();
-      return false;
-    }
-
-    return true;
-  },
-
-  /**
-   * Mark the viewed version of the beta appeal as the latest version
-   * stored in the schema.
-   */
-  markBetaAppealVersion() {
-    let currentBetaAppeal = Services.prefs.getIntPref(
-      "app.beta_appeal.version",
-      1
-    );
-    Services.prefs.setIntPref(
-      "app.beta_appeal.version.viewed",
-      currentBetaAppeal
-    );
-  },
 };
 
 /**
  * Called at startup to verify if we have ny existing account, even if invalid,
  * and if not, it will trigger the Account Hub in a tab.
  *
  * @returns {boolean} - True if we have at least one existing account.
  */
